### DISCUSSION TITLE
Datasource could be a free-form input box for custom renderizations

### DISCUSSION THREAD

**Discussion by dgonzalezruiz** [grafana/grafana#44138] [Dashboards, Panels & Visualizations]
The datasource entry on each panel currently limits the input to selecting between any existing instance datasource OR any `datasource` type variable defined locally by the dashboard.

I see the usecase of having N prometheus, each of them as `$datasource`, to be linked with another defined N loki `$datasource-loki` datasources --> for those scenarios, it would be nice being able to write `$datasource-loki` as the rendered datasource to select, so that by selecting a specific prometheus datasource you also can "get mapped" to the right loki datasource (in order to reduce the complexity to the dashboard user). However, this is currently only doable by editing the dashboard JSON model directly (which I have done, working successfully), not through UI (because of the input form limitations explained at beginning)

What do you think about changing that input type? Can you think of preferable, more efficient ways to map prometheus + loki datasources?

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
